In [1]:
!pip install mysql-connector-python

   ---------------------------------------- 0.0/17.7 MB ? eta -:--:--
   - -------------------------------------- 0.8/17.7 MB 14.5 MB/s eta 0:00:02
   -------- ------------------------------- 3.7/17.7 MB 13.2 MB/s eta 0:00:02
   ------------- -------------------------- 6.0/17.7 MB 12.6 MB/s eta 0:00:01
   ------------------ --------------------- 8.4/17.7 MB 12.0 MB/s eta 0:00:01
   ---------------------- ----------------- 10.0/17.7 MB 10.9 MB/s eta 0:00:01
   -------------------------- ------------- 11.8/17.7 MB 10.5 MB/s eta 0:00:01
   --------------------------- ------------ 12.1/17.7 MB 9.3 MB/s eta 0:00:01
   -------------------------------- ------- 14.4/17.7 MB 9.3 MB/s eta 0:00:01
   ------------------------------------- -- 16.5/17.7 MB 9.4 MB/s eta 0:00:01
   ---------------------------------------  17.6/17.7 MB 9.4 MB/s eta 0:00:01
   ---------------------------------------- 17.7/17.7 MB 8.6 MB/s  0:00:02



[notice] A new release of pip is available: 25.2 -> 26.1.2
[notice] To update, run: python.exe -m pip install --upgrade pip


In [2]:
import mysql.connector

print("MySQL Connector Installed Successfully")

MySQL Connector Installed Successfully


In [4]:
import mysql.connector

conn = mysql.connector.connect(
    host="localhost",
    user="amrutha",
    password="Amrutha@123",
    database="healthcare_db"
)

print("Connected Successfully")

Connected Successfully


In [6]:
import pandas as pd

df = pd.read_csv("heart_disease_data.csv")

print(df.shape)

df.head()

(606, 14)


,age,sex,cp,trestbps,chol,fbs,restecg,thalach,exang,oldpeak,slope,ca,thal,target
0,48,1,0,130,256,1,0,150,1,0.0,2,2,3,0
1,61,1,0,148,203,0,1,161,0,0.0,2,1,3,0
2,44,0,2,118,242,0,1,149,0,0.3,1,1,2,1
3,47,1,0,110,275,0,0,118,1,1.0,1,1,2,0
4,56,1,3,120,193,0,0,162,0,1.9,1,0,3,1


In [7]:
df.columns

Index(['age', 'sex', 'cp', 'trestbps', 'chol', 'fbs', 'restecg', 'thalach',
       'exang', 'oldpeak', 'slope', 'ca', 'thal', 'target'],
      dtype='object')

In [8]:
cursor = conn.cursor()

for _, row in df.iterrows():

    sql = """
    INSERT INTO patients (
        age, sex, cp, trestbps, chol,
        fbs, restecg, thalach, exang,
        oldpeak, slope, ca, thal, target
    )
    VALUES (%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s)
    """

    values = (
        int(row['age']),
        int(row['sex']),
        int(row['cp']),
        int(row['trestbps']),
        int(row['chol']),
        int(row['fbs']),
        int(row['restecg']),
        int(row['thalach']),
        int(row['exang']),
        float(row['oldpeak']),
        int(row['slope']),
        int(row['ca']),
        int(row['thal']),
        int(row['target'])
    )

    cursor.execute(sql, values)

conn.commit()

print("All records inserted successfully!")

All records inserted successfully!


In [9]:
print(df.shape)

(606, 14)


# SQL Analysis Report

## Objective

The objective of this phase was to store healthcare data in MySQL and perform SQL-based exploratory analysis.

## Key Findings

1. A total of 606 patient records were successfully loaded into the healthcare database.

2. SQL aggregation queries were used to analyze disease distribution across patients.

3. Average cholesterol and blood pressure levels were compared between disease and non-disease groups.

4. Gender-based analysis was performed to identify demographic trends.

5. High-risk patient groups were identified using cholesterol and blood pressure thresholds.

6. SQL enabled efficient healthcare data querying and supports future integration with Power BI dashboards.
